src/components/DatePicker/DatePicker.tsx

```diff
@@ -77,13 +79,19 @@ const DatePicker: React.FC<Props> = ({
   inputProps,
   dateFormatOverride = undefined,
   isClearable = false,
+  isDefaultNow = false,
 }) => {
+  const isValueUndefined = Object.values(value).every(v => !v);
+
   const [open, setOpen] = useState(false);
   const [selectedOption, setSelectedOption] = useState<string>('');
-  const [range, setRange] = useState<Range>({ from: dayjs(value.from), to: dayjs(value.to) });
+  const [range, setRange] = useState<Range>({
```

Comment:
you can move the state initialization into a function outside of the functional component.

example
```
const initDates = (shouldIncludeDates, newDates) => ({
    from: shouldIncludeDates ? dayjs(newDates.from) : undefined,
    to: shouldIncludeDates ? dayjs(newDates.to) : undefined,
})
```